### PR TITLE
add lighthouse (and devtools note) to crp docs

### PR DESCRIPTION
**Pull request by kaycebasques**
@igrigorik LGTY?

### CODE REVIEW

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 1]
```diff
@@ -5,20 +5,20 @@ description: Learn to measure the critical rendering path.
...
-- Navigation Timing provides high resolution timestamps for measuring CRP.
-- Browser emits series of consumable events which capture various stages of the CRP.
+## Instrumenting your code with the Navigation Timing API {: #navigation-timing }
 
-
-The foundation of every solid performance strategy is good measurement and instrumentation. Turns out, that is exactly what the Navigation Timing API provides.
+The foundation of every solid performance strategy is good measurement and
+instrumentation. The Navigation Timing API provides high-resolution timestamps
+for measuring CRP, and the browser emits a series of consumable events which
+capture various stages of the CRP.
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: Hmm, what do you think of rewording this a bit to highlight that NT is all about RUM tracking? Something like..

"The combination of Navigation Timing API, which provides a mechanism to gather high-resolution data for a number of important page load milestones, and other events emitted by the browser as the page is loading, enables us to gather and track the real-user performance of the critical rendering path."

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 2]
```diff
@@ -61,3 +61,26 @@ The above example may seem a little daunting on first sight, but in reality it i
 
 All said and done, we now have some specific milestones to track and a simple function to output these measurements. Note that instead of printing these metrics on the page you can also modify the code to send these metrics to an analytics server ([Google Analytics does this automatically](https://support.google.com/analytics/answer/1205784)), which is a great way to keep tabs on performance of your pages and identify candidate pages that can benefit from some optimization work.
 
+## Auditing a page with Lighthouse {: #lighthouse }
+
+Lighthouse is a web app auditing tool that runs a series of tests against a
+given page, and then displays the page's results in a consolidated report. You
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: Perhaps "local tests"? To help distinguish RUM vs local development..

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 3]
```diff
@@ -61,3 +61,26 @@ The above example may seem a little daunting on first sight, but in reality it i
...
+See [Auditing Web Apps With Lighthouse](/web/tools/lighthouse) to get started.
+
+## What about DevTools? {: #devtools }
+
+Although these docs sometimes use the Chrome DevTools Network panel to
+illustrate CRP concepts, DevTools is currently not well-suited for CRP
+measurements. The problem is that, as of October 2016, there is no
+filter for identifying critical resources. In other words, on a page
+that fetches tens or hundreds of resources, there is no way to filter
+out the non-critical resources.
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: Could we simplify this / avoid dates, etc?

"DevTools does not have a built-in mechanism to isolate the critical resources. To help identify such resources, try running a Lighthouse audit, or review guidance in [link to docs] to identify them manually."

**Comment by kaycebasques** [comment]
@igrigorik I updated per your feedback... have a looksie

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 4]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
-critical rendering path!
+The foundation of every solid performance strategy is good measurement and
+instrumentation. You can't optimize what you can't measure. This doc
+explains different approaches for measuring CRP performance.
 
-### TL;DR {: .hide-from-toc }
-- Navigation Timing provides high resolution timestamps for measuring CRP.
-- Browser emits series of consumable events which capture various stages of the CRP.
+The key difference between the different approaches is the different kind
+of metrics that they can capture.
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: are the different kinds?

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 5]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
 
+* The Lighthouse approach runs a series of automated tests against a page,
+  and then generates a report on the page's CRP performance. This approach
+  provides a quick and easy high-level overview of CRP performance, but does
+  not truly reflect how users experience the page.
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: This approach provides a quick and easy overview of the critical rendering path of a particular page loaded in your browser, allowing you to rapidly test, iterate, and improve its performance.

Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: Scratch the "but does not...".

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 6]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
+* The Navigation Timing API approach captures [Real User
+  Monitoring (RUM)](https://en.wikipedia.org/wiki/Real_user_monitoring)
+  metrics. As the name implies, these metrics are captured from real user
+  interactions with your site. This approach provides a more accurate view
+  of CRP performance, but is more difficult and time-consuming to set up.
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: "with your site, which provides an accurate view into the real-world CRP performance, as experienced your users across a variety of devices and network conditions". 

Scratch the but is more difficult..

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 7]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
 
-The foundation of every solid performance strategy is good measurement and instrumentation. Turns out, that is exactly what the Navigation Timing API provides.
+In general, a good approach is to use Lighthouse to identify obvious CRP
+optimization opportunities, and then to instrument your code with the
+Navigation Timing API to truly measure how your app performs out in the wild.
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: s/to truly measure/to monitor/

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 8]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
+
+## Auditing a page with Lighthouse {: #lighthouse }
+
+Lighthouse is a web app auditing tool that runs a series of tests against a
+given page, and then displays the page's results in a consolidated report. You
+can run Lighthouse as a Chrome Extension or NPM module (which is
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: drop the () and use a comma?

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 9]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
+useful for integrating Lighthouse with continuous integration systems).
+
+Lighthouse has a suite of audits for measuring a page's CRP. When you
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: Drop the first sentence? Seems like we're repeating ourselves.

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 10]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
+run Lighthouse as a Chrome Extension, your page's CRP results look like the
+screenshot below.
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: Can we elaborate on what the screenshot shows? E.g. call out longest chain, etc.

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md [hunk 11]
```diff
@@ -5,20 +5,51 @@ description: Learn to measure the critical rendering path.
...
+
+![Lighthouse's CRP audits](images/lighthouse-crp.png)
+
+See [Auditing Web Apps With Lighthouse](/web/tools/lighthouse) to get started.
+
+## Instrumenting your code with the Navigation Timing API {: #navigation-timing }
+
+The Navigation Timing API provides high-resolution timestamps
```
Comment (igrigorik) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: How about.. "The combination of Navigation Timing API and other browser events emitted as the page is loaded allows us to capture and record real-world CRP performance of any page. "

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md (hunk 4 above)
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: Nope. The verb should match the subject of the sentence, which is "the key difference". The way to test these scenarios is by removing the extra element ("the key difference... is").

This sentence is funky though so I'll reword it.

File: src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md (hunk 10 above)
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/critical-rendering-path/measure-crp.md]: I added a link to the audit's reference doc, which is merged and should be live soon. We can wait until then before merging this.

Here's the reference doc source which you can review in the meantime: https://github.com/google/WebFundamentals/pull/3693/files#diff-45fb923d6ae73708ae68413d31216d4c

**Comment by kaycebasques** [comment]
@igrigorik take three ;)

**Comment by kaycebasques** [comment]
@petele good to go

**Pull request merged**